addons/microsoft_calendar/models/calendar_recurrence_rule.py

```diff
@@ -138,7 +138,8 @@ def _get_microsoft_sync_domain(self):
         # older versions of the module. When synced, these recurrence may come back from Microsoft after database cleaning
         # and trigger errors as the records are not properly populated.
         # We also prevent sync of other user recurrent events.
-        return [('calendar_event_ids.user_id', '=', self.env.user.id), ('rrule', '!=', False)]
+        return [('calendar_event_ids.user_id', '=', self.env.user.id), ('rrule', '!=', False),
```

Comment:
I fixed it from `_get_microsoft_sync_domain` because when you try to sync a user it would call `restart_microsoft_synchronization` , and it makes the `need_sync_m` = true based on that domain.